Engine/platform/osx/acplmac.cpp

```diff
@@ -113,6 +151,31 @@ void AGSMac::ShutdownCDPlayer() {
   //cd_exit();
 }
 
+const char* AGSMac::GetAllUsersDataDirectory()
+{
+  return commonDataPath;
+}
+
+const char *AGSMac::GetUserSavedgamesDirectory()
+{
+  return libraryApplicationSupport;
+}
+
+const char *AGSMac::GetUserConfigDirectory()
+{
+  return libraryApplicationSupport;
+}
+
+const char *AGSMac::GetAppOutputDirectory()
+{
+  return commonDataPath;
+}
+
+const char *AGSMac::GetIllegalFileChars()
+{
+  return "\\/:?\"<>|*"; // keep same as Windows so we can sync.
```

Comment:
Would you like to do that in this pull request or create a task? I'm not entirely familiar with the filename code so I would need to spend some time on that.